cmd/kubelet/app/server.go

```diff
@@ -1169,25 +1169,18 @@ func RunKubelet(kubeServer *options.KubeletServer, kubeDeps *kubelet.Dependencie
 		return fmt.Errorf("failed to create kubelet: %w", err)
 	}
 
-	// NewMainKubelet should have set up a pod source config if one didn't exist
```

Comment:
moved this error handling into the `NewMainKubelet` flow itself, close to the source of such errata